src/main/scala/uk/gov/nationalarchives/fileformat/Lambda.scala

```diff
@@ -2,41 +2,56 @@ package uk.gov.nationalarchives.fileformat
 
 import com.amazonaws.services.lambda.runtime.Context
 import com.amazonaws.services.lambda.runtime.events.SQSEvent
+import com.amazonaws.services.lambda.runtime.events.SQSEvent.SQSMessage
 import com.typesafe.config.{Config, ConfigFactory}
-import software.amazon.awssdk.services.sqs.model.DeleteMessageResponse
+import io.circe.Decoder
+import io.circe.generic.auto._
+import io.circe.generic.semiauto.deriveDecoder
+import io.circe.parser.decode
+import software.amazon.awssdk.services.sqs.model.{DeleteMessageResponse, SendMessageResponse}
 import uk.gov.nationalarchives.aws.utils.Clients.sqs
-import uk.gov.nationalarchives.aws.utils.S3EventDecoder._
 import uk.gov.nationalarchives.aws.utils.SQSUtils
+import uk.gov.nationalarchives.fileformat.FFIDExtractor.FFIDFile
 
-import scala.concurrent.ExecutionContext.Implicits.global
-import scala.concurrent.duration._
-import scala.concurrent.{Await, Future}
 import scala.jdk.CollectionConverters._
 import scala.language.postfixOps
+import uk.gov.nationalarchives.fileformat.FFIDExtractor._
 
 class Lambda {
 
+  case class FFIDFileWithReceiptHandle(ffidFile: FFIDFile, receiptHandle: String)
+
   val config: Config = ConfigFactory.load
   val sqsUtils: SQSUtils = SQSUtils(sqs)
 
   val deleteMessage: String => DeleteMessageResponse = sqsUtils.delete(config.getString("sqs.queue.input"), _)
+  val sendMessage: String => SendMessageResponse = sqsUtils.send(config.getString("sqs.queue.output"), _)
+
+  val downloadOutput: Decoder[FFIDFile] = deriveDecoder[FFIDFile].map[FFIDFile](identity)
+
 
   def process(event: SQSEvent, context: Context): List[String] = {
-    val eventsWithErrors: EventsWithErrors = decodeS3EventFromSqs(event)
-    val fileUtils = FileUtils()
-    val recordProcessor = RecordProcessor(sqsUtils, fileUtils)
-    val processingResult: List[Future[Either[String, String]]] = eventsWithErrors.events
-          .flatMap(e => e.event.getRecords.asScala
-          .map(r => recordProcessor.processRecord(r, e.receiptHandle)))
-
-    val receiptHandleOrError = Await.result(Future.sequence(processingResult), 1000 seconds)
-    val (fileFormatFailed: List[String], fileFormatSucceeded: List[String]) = receiptHandleOrError.partitionMap(identity)
-    val allErrors = fileFormatFailed ++ eventsWithErrors.errors.map(_.getCause.getMessage)
-    if (allErrors.nonEmpty) {
-      fileFormatSucceeded.foreach(deleteMessage)
-      throw new RuntimeException(allErrors.mkString("\n"))
+    def extractFFID(fileWithHandle: FFIDFileWithReceiptHandle): Either[String, String] = {
```

Comment:
Yeah, I agree. I've changed the `logAndSummarise` method to return and `ErrorSummary` case class which takes the message and the original exception. I've moved the logging to the end of the lambda function as well because we've now got the original exception there.